src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @fileoverview This rule checkes header strict-transport-security is sent for resources served over HTTPS, and max-age has proper  value.
+ */
+
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+import { isDataURI, isHTTPS, normalizeString } from '../../utils/misc'; // eslint-disable-line no-unused-vars
+import { IAsyncHTMLElement, IResponse, IFetchEnd, IRule, IRuleBuilder } from '../../types'; // eslint-disable-line no-unused-vars
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The minimum period (in seconds) allowed for `max-age`. */
+        let minimum: number;
+
+        const loadRuleConfigs = () => {
+            minimum = (context.ruleOptions && context.ruleOptions.minimum) || 10886400;
+        };
+
+        const maxAgeLessThan = (headerValue: string, limit: number) => {
+            const maxAge = headerValue.match(/max-age=([^;]+);|max-age=([^;]+)$/).pop();
```

Comment:
A header value parser would be great, but...

---

I think this will fail for something such as `Max-AGE=123456789`. Also, wouldn't it be more easy and clear to use `split`?

See also: https://tools.ietf.org/html/rfc6797#section-6.1